src/main/java/roomescape/model/Reservation.java

```diff
@@ -0,0 +1,86 @@
+package roomescape.model;
+
+import java.time.LocalDate;
+
+public class Reservation {
+
+    private final Long id;
+    private final String name;
+    private final LocalDate date;
+    private final ReservationTime reservationTime;
+    private final Theme theme;
+
+    public Reservation(Long id, String name, LocalDate date, ReservationTime reservationTime, Theme theme) {
+        validateRequiredFields(id, name, date, reservationTime, theme);
+
+        this.id = id;
+        this.name = name;
+        this.date = date;
+        this.reservationTime = reservationTime;
+        this.theme = theme;
+    }
+
+    public Reservation(String name, LocalDate date, ReservationTime reservationTime, Theme theme) {
+        validateRequiredFields(name, date, reservationTime, theme);
+
+        this.id = null;
+        this.name = name;
+        this.date = date;
+        this.reservationTime = reservationTime;
+        this.theme = theme;
+    }
```

Comment:
이렇게 매 생성자 마다 값을 세팅하기보다 세팅을 위한 목적의 기본 생성자를 두고 내부 호출 통해 개선도 가능할 것 같아요!
```suggestion
    public Reservation(Long id, String name, LocalDate date, ReservationTime reservationTime, Theme theme) {
        validateRequiredFields(id, name, date, reservationTime, theme);

        this.id = id;
        this.name = name;
        this.date = date;
        this.reservationTime = reservationTime;
        this.theme = theme;
    }

    public Reservation(String name, LocalDate date, ReservationTime reservationTime, Theme theme) {
        this(null, name, date, reservationTime, theme);
    }
```

Reply:
제안 감사합니다 매트!

제가 기존의 코드와 같이 작성했던 이유는 파라미터로 id 가 들어온 경우에는 id 가 null 인지에 대한 검증이 필요해서였는데요, 이 부분은 [이 코멘트](https://github.com/woowacourse/spring-roomescape-member/pull/204#discussion_r2072333270) 에서의 판단에 따라 달라질 것 같아 추후에 다시 리팩토링 해봐도 괜찮을까요?! 🙇🏻

Reply:
넵 좋습니다! 아래 질문에 답변 남겨둘게요 ㅎㅎ